### PR TITLE
Migrate to ESLint 9.1.1

### PR DESCRIPTION
**Pull request by basil**
See:

https://eslint.org/docs/latest/use/migrate-to-9.0.0
https://eslint.org/docs/latest/use/configure/migration-guide

### Testing done

Ran `yarn lint:js` locally. Also introduced some errors in both `war/` and `core/` and verified they were fixed with `yarn lint:fix`.

### Proposed changelog entries

N/A

### Proposed upgrade guidelines

N/A

```[tasklist]
### Submitter checklist
- [ ] The Jira issue, if it exists, is well-described.
- [ ] The changelog entries and upgrade guidelines are appropriate for the audience affected by the change (users or developers, depending on the change) and are in the imperative mood (see [examples](https://github.com/jenkins-infra/jenkins.io/blob/master/content/_data/changelogs/weekly.yml)). Fill in the **Proposed upgrade guidelines** section only if there are breaking changes or changes that may require extra steps from users during upgrade.
- [ ] There is automated testing or an explanation as to why this change has no tests.
- [ ] New public classes, fields, and methods are annotated with `@Restricted` or have `@since TODO` Javadocs, as appropriate.
- [ ] New deprecations are annotated with `@Deprecated(since = "TODO")` or `@Deprecated(forRemoval = true, since = "TODO")`, if applicable.
- [ ] New or substantially changed JavaScript is not defined inline and does not call `eval` to ease future introduction of Content Security Policy (CSP) directives (see [documentation](https://www.jenkins.io/doc/developer/security/csp/)).
- [ ] For dependency updates, there are links to external changelogs and, if possible, full differentials.
- [ ] For new APIs and extension points, there is a link to at least one consumer.
```

### Desired reviewers

@mention

Before the changes are marked as `ready-for-merge`:

```[tasklist]
### Maintainer checklist
- [ ] There are at least two (2) approvals for the pull request and no outstanding requests for change.
- [ ] Conversations in the pull request are over, or it is explicit that a reviewer is not blocking the change.
- [ ] Changelog entries in the pull request title and/or **Proposed changelog entries** are accurate, human-readable, and in the imperative mood.
- [ ] Proper changelog labels are set so that the changelog can be generated automatically.
- [ ] If the change needs additional upgrade steps from users, the `upgrade-guide-needed` label is set and there is a **Proposed upgrade guidelines** section in the pull request title (see [example](https://github.com/jenkinsci/jenkins/pull/4387)).
- [ ] If it would make sense to backport the change to LTS, a Jira issue must exist, be a _Bug_ or _Improvement_, and be labeled as `lts-candidate` to be considered (see [query](https://issues.jenkins.io/issues/?filter=12146)).
```

### CODE REVIEW

File: war/package.json
Comment (basil) [inline war/package.json:22]: Reversing the order for parity with `lint:js`.

File: war/package.json
Comment (basil) [inline war/package.json:29]: Detached into a separate package.

File: war/package.json
Comment (basil) [inline war/package.json:36]: Detached into a separate package.

File: war/src/main/js/filter-build-history.js
Comment (basil) [inline war/src/main/js/filter-build-history.js:330]: Now prints a new warning that we're suppressing a nonexistent warning.

File: war/src/main/js/pluginSetupWizardGui.js
Comment (basil) [inline war/src/main/js/pluginSetupWizardGui.js:291]: Suppressing a new warning.

**Comment by basil** [comment]
This PR, as-is, somehow makes ESLint stop checking parent directories, as I found when intentionally introducing an error in

```
diff --git a/core/src/main/resources/lib/form/password/password.js b/core/src/main/resources/lib/form/password/password.js
index 4cd33dbfc6..70d24f77db 100644
--- a/core/src/main/resources/lib/form/password/password.js
+++ b/core/src/main/resources/lib/form/password/password.js
@@ -28,9 +28,8 @@ Behaviour.specify(
   0,
   function (e) {
     var secretUpdateBtn = e.querySelector(".hidden-password-update-btn");
-    if (secretUpdateBtn === null) {
+    if (secretUpdateBtn === null)
       return;
-    }
 
     secretUpdateBtn.onclick = function () {
       e.querySelector(".hidden-password-field").setAttribute(
```

Before this PR, this error would be caught, but with this PR it isn't caught anymore.

Perhaps someone like @NotMyFault or @timja can push a commit that fixes this so that this PR can be taken out of draft state.

**Comment by timja** [comment]
> Perhaps someone like @NotMyFault or @timja can push a commit that fixes this so that this PR can be taken out of draft state.

Likely this from the migration guide: https://eslint.org/docs/latest/use/configure/migration-guide#additional-changes

**Comment by basil** [comment]
Yeah that seems plausible. It would be great if someone could push a commit that fixes this so that this PR can be taken out of draft state.

**Comment by basil** [comment]
Thanks @timja ! I have marked this PR as ready for review

**Review by timja — APPROVED** [review]
Thanks!

/label ready-for-merge

----

This PR is now ready for merge, after ~24 hours, we will merge it if there's no negative feedback.

Thanks!

**Comment by janfaracik** [comment]
I might be doing something dumb - 

```
Error: Cannot find module 'eslint-config-prettier'
Require stack:
- /Users/janfaracik/Developer/jenkins/eslint.config.cjs
Require stack:
- /Users/janfaracik/Developer/jenkins/eslint.config.cjs
    at Module._resolveFilename (node:internal/modules/cjs/loader:1143:15)
    at require$$0.Module._resolveFilename (/Users/janfaracik/Developer/jenkins/war/.pnp.cjs:15910:54)
    at Module._load (node:internal/modules/cjs/loader:984:27)
    at require$$0.Module._load (/Users/janfaracik/Developer/jenkins/war/.pnp.cjs:15824:31)
    at Module.require (node:internal/modules/cjs/loader:1231:19)
    at require (node:internal/modules/helpers:179:18)
    at Object.<anonymous> (/Users/janfaracik/Developer/jenkins/eslint.config.cjs:1:30)
    at Module._compile (node:internal/modules/cjs/loader:1369:14)
    at Module._extensions..js (node:internal/modules/cjs/loader:1427:10)
    at require$$0.Module._extensions..js (/Users/janfaracik/Developer/jenkins/war/.pnp.cjs:15976:33)
```

Getting this when running anything lint related, e.g. `yarn lint:fix`

**Comment by timja** [comment]
You've ran `yarn install?`

**Comment by janfaracik** [comment]
> You've ran `yarn install?`

Yeah I did.

**Comment by timja** [comment]
I get the same issue locally =/

**Comment by timja** [comment]
This works:

```
mvn -pl war frontend:yarn -Dfrontend.yarn.arguments=lint:fix
```

(linked in CONTRIBUTING guide https://github.com/jenkinsci/jenkins/blob/master/CONTRIBUTING.md#linting)

**Comment by timja** [comment]
I suspect its something to do with working directory, the config file is above the package.json path.

**Comment by timja** [comment]
I tried a few things and can't see a fix except for one of:

1. Creating a package.json in the root
2. Creating a package.json in any module that needs eslint

**Pull request merged**
